src/main/java/com/zufar/icedlatte/security/endpoint/UserSecurityEndpoint.java

```diff
@@ -44,15 +45,20 @@ public class UserSecurityEndpoint implements SecurityApi {
     private final ChangeUserPasswordOperationPerformer changeUserPasswordOperationPerformer;
 
     private final HttpServletRequest httpRequest;
+    private final EmailValidator emailValidator;
 
     @Override
     @PostMapping("/register")
     public ResponseEntity<String> register(@RequestBody final UserRegistrationRequest request) {
-        log.info("Received registration request for user with email = '{}'", request.getEmail());
+        String email = request.getEmail();
+        log.info("Received registration request for user with email = '{}'", email);
+        if (emailValidator.isNotValid(email) && !emailValidator.isEmailUnique(email)) {
+            return ResponseEntity.badRequest().body("Invalid email format");
```

Comment:
nit: Hm, we usually throw exception and add [a handler](https://github.com/Sunagatov/Iced-Latte/blob/fed96817fbdcd42d8d650e0695e7788adbb735e9/src/main/java/com/zufar/icedlatte/email/exception/EmailExceptionHandler.java#L60) which returns "Bad request" 🤔

Reply:
UPD: Haven't noticed initially, that you delegated that to Open API, got it 👍